src/execution/sql/thread_state_container.cpp

```diff
@@ -47,7 +47,8 @@ ThreadStateContainer::TLSHandle::~TLSHandle() {
 
 // The actual container for all thread-local state for participating threads
 struct ThreadStateContainer::Impl {
-  tbb::enumerable_thread_specific<TLSHandle> states_;
+  common::SpinLatch states_latch_;
+  std::map<std::thread::id, TLSHandle *> states_;
```

Comment:
Why `std::map` and not `std::unordered_map` ?

Reply:
Also, why not use `std::unique_ptr<>` for the values? That should simplify your cleanup logic.

Reply:
1) std::map is giving better performance in the benchmarks as compared to std::unordered_map.
2) We did use unique_ptr but that gave us other clang-tidy issues.

Reply:
1) std::map consistently performed better in the benchmarks. I am not able to reason this out though.
2) We did try this but it gave us other clang-tidy issues and so we moved away from it.

Reply:
Do you recall what `clang-tidy` issues came up with `std::unique_ptr`? Apply the changes here and let's see what happens.